### ISSUE TITLE
Implement Organization Edit Form Tooltip Re-Design

### ISSUE THREAD

**Issue by entrotech** [hackforla/food-oasis#1588]
### Overview
The UI/UX team came up with a re-design of the tooltips on the Organization Edit page that we need to implement. It is believed to improve the User Experience for the form.

### Action Items
- [x] - Review Design Issue #1148 and clarify any questions.
- [ ] Implement the changes as part of standardizing the Text Edit Field design. This should apply to all Form Controls (TextFields, Check Boxes, Radio Buttons, Drop-Downs, and Buttons.
- [ ] This issue may need to be broken up into separate issues for the various control types.

### Resources/Instructions
- Issue #1148

**Comment by andyarensman** [comment]
![chrome_pUN3ooBomd](https://user-images.githubusercontent.com/86622025/234157801-28f7cd93-3e5c-4923-b360-6de131b38af5.png)
![chrome_JU5QLbNcOZ](https://user-images.githubusercontent.com/86622025/234157980-a09bda90-4686-4393-9e89-14a25eca040e.png)

When the user clicks the icon, the tooltip appears. The user can close it by clicking the icon again, or clicking anywhere else.

The new labels were added. This will cause some temporary visual issues where there are other textfields, but these will be resolved soon with a new card.

**Comment by fancyham** [comment]
Looking good — would it be possible to move the bubbles downwards so that they’re closer to the (i) symbol, or alternatively, to make the triangle bigger so that it is further down?  (see mockups in original issue)   That would make it feel more related/attached to the info button icon — they’re a bit far away right now.

<img width="1006" alt="image" src="https://github.com/hackforla/food-oasis/assets/3376957/59c6d570-8140-4be7-a9b5-b8ae86c3e00e">

also, question: is it trivial to put a bit of a drop shadow under the tooltip, also, a slight outline for the tooltip itself?  If those are easy, I might be interested to see what that might look like (some examples are in the original issue’s comments)… I think we could use a little dimensionality.  But if they’re hard, then what we have is already a huge improvement.

**Comment by andyarensman** [comment]
@fancyham Here is what it would look like with an outline:
![chrome_ggBmXYZe0H](https://github.com/hackforla/food-oasis/assets/86622025/69410d90-cacb-444d-8384-1c7ebd09ba69)
You can see that the outline does not go around the arrow so it looks a little weird. The same thing happens with the shadow.

Here is what it looks like without a border now, with the lower position:
![chrome_YDU01Ej8O1](https://github.com/hackforla/food-oasis/assets/86622025/464fabdd-ed10-44ad-bdd0-6c246f1ec5ba)
This will apply to all the tooltips once it is pushed.

**Comment by fancyham** [comment]
Lemme ask the design team if they have a preference — thanks for the screenshots!

**Comment by fancyham** [comment]
Also, would it be possible to do a drop shadow?  Perhaps that shadow would include the triangle?

**Comment by andyarensman** [comment]
The drop shadow does not include the triangle. 
![chrome_7yHknAXGcM](https://github.com/hackforla/food-oasis/assets/86622025/d7adfde9-9c52-4375-878d-4a42c6b8899a)

**Comment by fancyham** [comment]
Quick Update: 

TL;DR: deliberation still in progress… trying to come up with more contrast/easily readable text.

Dev was able to match our original design... during QA, I asked if we could see versions with an outline and a drop shadow in an attempt to up the contrast/readability..    Unfortunately, those two don’t work for the triangle portion of the tool-tip. (shown above)

So, started a conversation with designers those options and none came out as the clear best… so some further discussion in the #fola-design channel, we talked about legibility.  I pulled some other variations that have been considered before, and how they looked:

From w3schools [website](https://www.w3schools.com/css/css_tooltip.asp):
![image](https://github.com/hackforla/food-oasis/assets/3376957/62fad963-09fa-4333-8516-dfe3c715bb21)

From an earlier design in the [tooltip design issue #1148](https://github.com/hackforla/food-oasis/issues/1148#issuecomment-1043725104)
![image](https://github.com/hackforla/food-oasis/assets/3376957/c63f6fe4-5457-4bfb-bd62-2b92e7c2d6a8)

The translucency was making it hard to read, so here’s a version that's opaque that we're [discussing](https://hackforla.slack.com/archives/CPTLCM9J4/p1684102516431639):
![image](https://github.com/hackforla/food-oasis/assets/3376957/f1b92a0d-3b3d-4751-bb6e-f08592009f42)

deliberation still in progress… thanks for your patience

**Comment by andyarensman** [comment]
I ended up figuring out how to add the drop shadow to the arrow and the rest of the tooltip. It's a little finicky, but this is what it looks like now:
<img width="437" alt="Screen Shot 2023-05-16 at 6 15 25 PM" src="https://github.com/hackforla/food-oasis/assets/86622025/45c690af-4db6-43e3-88a1-ce0fa446b181">

**Comment by fancyham** [comment]
Oh, interesting. Is it robust and easy enough for us to use?  Or fragile and hard to get right for all the various tooltip possibilities?

**Comment by andyarensman** [comment]
It's just the arrow portion that is challenging. There's a weird box around it that will cut off the shadow if the shadow gets too big, like this (made red to be more noticeable):
![chrome_LpdVEopgQm](https://github.com/hackforla/food-oasis/assets/86622025/1fa5dd3e-1b1a-48f0-be29-adf6201c2122)
So the shadow I have in my previous comment is the best I could position it. The shadow gets a little dark where the arrow overlaps with the rectangle shadow and there doesn't seem to be a way around this, but you can't really tell. **Changing the shape and size of the arrow itself would be very challenging if possible at all, so I recommend leaving the shape and size alone.**

If you want to change the shadow of just the rectangle, that's doable.

Also, I am now able to add a border to the full tooltip with the new technique I found. Here is an example:
 ![chrome_zEbQuU6Hzd](https://github.com/hackforla/food-oasis/assets/86622025/3f326e4d-f349-42a1-a5be-eb9bc62e09b2)

**Comment by fancyham** [comment]
Nice. That outline is pretty hard to do, I bet. Is that working well for all the browsers?  If so, I can bring them as options for the designers. I know it can be tricky to do some of these kinds of things with multiple shapes.

**Comment by andyarensman** [comment]
It looks good to me in Chrome, Safari, and Edge. It's easy to change the thickness and color of the outline now that I have it set up.

**Comment by fancyham** [comment]
Cool — just talked with the design team and right now, it's a tie between the dark solid tooltip with white text, and the drop shadow one.  They make it pop more while being legible and looking good.   I'm gonna solicit a bit more feedback on Slack to help choose. :)

**Comment by fancyham** [comment]
Hi Andy, could you take some zoomed-out screenshots of the latest dropshadow version but without the red highlight for two situations that recreates these two tooltip mockups?

![image](https://github.com/hackforla/food-oasis/assets/3376957/e3535d07-0d34-479a-aa7c-8e899d234313)

I’d like to use that for the designers to better be able to compare the two side-by-side with the dark background-white text version. 

Those two are the favorites — and for designs, it’s best to see it in context of other elements on the screen.

Thanks!

**Comment by andyarensman** [comment]
Sure thing! These are currently live on the devla site if they want to see more context.

Mobile:
![chrome_Y5hHiKydaw](https://github.com/hackforla/food-oasis/assets/86622025/49286334-328f-4f07-a015-f9b8e6e573bf)

Desktop:
![chrome_nXBPwMwQyW](https://github.com/hackforla/food-oasis/assets/86622025/310306d8-7652-4df8-a5f6-803c187613c2)

**Comment by fancyham** [comment]
Thanks, Andy, for all the variations and screenshots.  The design team felt the tooltip needed more differentiation from the background, so after two rounds of variations and voting, the drop shadow version came out ahead by a good margin, so let’s go with that!

![image](https://github.com/hackforla/food-oasis/assets/3376957/763ef1dc-2958-4609-90fd-c00bf2efe247)

**Comment by andyarensman** [comment]
Great! That is the version currently deployed on the development site, so I will go ahead and close this issue.
